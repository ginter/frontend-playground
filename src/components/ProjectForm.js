import React from 'react'

const ProjectForm = ({ name = '', onSubmit }) => {
  let _name;

  return(
    <form name='new-project' onSubmit={e => {
      e.preventDefault();
      onSubmit(_name.value);
      _name.value = ''; }}>
      <input
        type='text'
        name='project-name'
        defaultValue={name}
        ref={c => _name = c }
        onChange={e => _name.value = e.target.value} />
      <input type='submit' />
    </form>
  );
}

export default ProjectForm
