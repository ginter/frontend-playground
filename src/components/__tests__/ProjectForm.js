import React from 'react'
import renderer from 'react-test-renderer'

import ProjectForm from '../ProjectForm'

describe('ProjectForm', () => {
  it('renders correctly with no name provided', () => {
    const component = renderer.create(<ProjectForm />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with a name provided', () => {
    const component = renderer.create(<ProjectForm name='name' />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
