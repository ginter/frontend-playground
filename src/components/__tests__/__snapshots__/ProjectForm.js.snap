exports[`ProjectForm renders correctly with a name provided 1`] = `
<form
  name="new-project"
  onSubmit={[Function onSubmit]}>
  <input
    defaultValue="name"
    name="project-name"
    onChange={[Function onChange]}
    type="text" />
  <input
    type="submit" />
</form>
`;

exports[`ProjectForm renders correctly with no name provided 1`] = `
<form
  name="new-project"
  onSubmit={[Function onSubmit]}>
  <input
    defaultValue=""
    name="project-name"
    onChange={[Function onChange]}
    type="text" />
  <input
    type="submit" />
</form>
`;
